### PR TITLE
GEODE-5187: clients can miss events when servers recycled

### PR DESCRIPTION
**Pull request by ladyVader**
* When draining events from the giiQueue, the msg field of HAEventWrapper may be null.
  Update the HAEventWrapper to point to the message in the HAContainer vs. calling
  putEventInHARegion with the original HAContainer message (a ClientUpdateMessageImpl).
  This is necessary as the ClientUpdateMessageImpl does not have the eventId (this is not
  serialized/deserialized in toData/fromData).  The HAEventWrapper is required on the
  remote side to reconstruct the event.

* Updated log messages to include the HARegionQueue.regionName

* Added corresponding IntegrationTest

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java [hunk 1]
```diff
@@ -3474,12 +3478,14 @@ protected void putEventInHARegion(Conflatable event, Long position) {
       }
       // Put the reference to the HAEventWrapper instance into the
       // HA queue.
+      // logger.info("adding inputHaEventWrapper to HARegion at " + position + ":" +
```
Comment (nabarunnag) [inline geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java]: Should we remove these commented out code?

File: geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java [hunk 2]
```diff
@@ -3474,12 +3478,14 @@ protected void putEventInHARegion(Conflatable event, Long position) {
...
+      // inputHaEventWrapper + " for " + this.regionName);
```
Comment (boglesby) [inline geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java]: Do you want to remove this commented out info message or convert it to a debug?

File: geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java [hunk 3]
```diff
@@ -3474,12 +3478,14 @@ protected void putEventInHARegion(Conflatable event, Long position) {
...
       this.region.put(position, inputHaEventWrapper);
-      // logger.info(LocalizedStrings.DEBUG, "added message at position " + position);
     } else { // (event instanceof ClientMarkerMessageImpl OR ConflatableObject OR
              // ClientInstantiatorMessage)
+      // logger.info("adding ClientUpdateMessage to HARegion at " + position + ":" + event + " for "
+      // + this.regionName);
```
Comment (boglesby) [inline geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java]: Do you want to remove this commented out info message or convert it to a debug?

File: geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java (hunk 2 above)
Comment (ladyVader) [reply geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java]: converting to debug

File: geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java (hunk 3 above)
Comment (ladyVader) [reply geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java]: converting to debug

**Pull request merged**